picky/src/ssh/private_key.rs

```diff
@@ -173,6 +205,17 @@ impl SshPrivateKey {
         &self.base_key
     }
 
+    pub fn inner_key(&self) -> Result<&PrivateKey, SshPrivateKeyError> {
+        match self.base_key() {
+            SshBasePrivateKey::Rsa(key) => Ok(key),
+            SshBasePrivateKey::Ec(key) => Ok(key),
+            SshBasePrivateKey::Ed(key) => Ok(key),
+            SshBasePrivateKey::SkEcdsaSha2NistP256 { .. } | SshBasePrivateKey::SkEd25519 { .. } => {
+                Err(SshPrivateKeyError::SkKeyDoesNotContainPrivateKeyData)
```

Comment:
If a function can only fail in one case, it makes sense to return the value as `Option<...>` instead of `Result<..., ...>`

Reply:
Agreed!

Reply:
Done